### PR TITLE
Add FlatBuffers for `{ewah_,null_,wah_,}bitmap`

### PR DESCRIPTION
**Pull request by dominiklohmann**
For now, this requires copying data out of the FlatBuffers tables, but they are designed in a way that allows for creating zero-copy views that only hold a `flatbuffer<T>` in the future.

This is not yet used in practice, because that requires a larger change. I've decided to split this up into multiple PRs for improved reviewability.

### :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

File-by-file.

~This still needs unit tests, which I plan to add in a second commit.~

### CODE REVIEW

File: libvast/vast/fbs/bitmap.fbs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+namespace vast.fbs.bitmap.detail;
+
+table BitVector {
+  blocks: [ulong] (required);
+  size: ulong;
+}
+
+namespace vast.fbs.bitmap;
+
+table EwahBitmap {
```
Comment (mavam) [inline libvast/vast/fbs/bitmap.fbs:10]:
```suggestion
table EWAHBitmap {
```

File: libvast/vast/fbs/bitmap.fbs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  blocks: [ulong] (required);
+  last_marker: ulong;
+  num_bits: ulong;
+}
+
+table NullBitmap {
+  bit_vector: detail.BitVector (required);
+}
+
+table WahBitmap {
```
Comment (mavam) [inline libvast/vast/fbs/bitmap.fbs:20]:
```suggestion
table WAHBitmap {
```

**Review by mavam — COMMENTED** [review]
Just a comment on the naming of acronyms.

File: libvast/vast/bitmap.hpp [hunk 3]
```diff
@@ -107,7 +114,6 @@ bitmap_bit_range bit_range(const bitmap& bm);
 } // namespace vast
 
 namespace caf {
-
```
Comment (tobim) [inline libvast/vast/bitmap.hpp:110]: nit: why delete this line?

File: libvast/src/bitmap.cpp [hunk 4]
```diff
@@ -57,6 +89,60 @@ bool operator==(const bitmap& x, const bitmap& y) {
...
+  switch (from.bitmap_type()) {
+    case fbs::bitmap::Bitmap::NONE:
+      return caf::make_error(ec::logic_error, "invalid vast.fbs.Bitmap type");
+    case fbs::bitmap::Bitmap::ewah: {
+      const auto& from_ewah = *from.bitmap_as_ewah();
+      auto to_ewah = ewah_bitmap{};
+      if (auto err = unpack(from_ewah, to_ewah))
+        return err;
+      to.bitmap_ = std::move(to_ewah);
+      return caf::none;
```
Comment (tobim) [inline libvast/src/bitmap.cpp:124]: I think it should be possible to define a lambda that offers an API like:
```c++
return do_unpack(*from.bitmap_as_ewah(), ewah_bitmap{});
```
And use that for every case.

**Review by tobim — APPROVED** [review]
The `bitmap.fbs` file doesn't contain any surprises, which is great.

File: libvast/vast/bitmap.hpp (hunk 3 above)
Comment (dominiklohmann) [reply libvast/vast/bitmap.hpp]: Probably by mistake, I added it back.

File: libvast/src/bitmap.cpp (hunk 4 above)
Comment (dominiklohmann) [reply libvast/src/bitmap.cpp]: Good idea, I've pushed that in a separate commit.

**Pull request merged**